README.md
# Sagiri-Discord.py-Bot
Aio Discord.py bot | work in progress

Hello to everyone,
this is my first Discord bot. The bot is based on python,
it has some moderation, gambling and music commands along with the standart ping command and a custom help command

I am gonna add more features in the future
Have fun!


Changelog.txt
Sagiri Discord.py Bot
    Developed by Senpai#0001

Version 1.0(November 14th, 2021)
    -custom prefixes
    -moderation
    -gambling
    -ping
    #pls don't ask | it's just a thing between me and my friends
        -three fun comments

Version 1.1(November 16th, 2021)
    -music command
    -some little improvements

sagiri.py
import os
import discord
from dotenv import load_dotenv
from discord.ext import commands
import json
import logging
from extensions import maintenance, moderation, gambling, fun, music

load_dotenv()

TOKEN = os.getenv("DISCORD_TOKEN")

logger = logging.getLogger('discord')
logger.setLevel(logging.DEBUG)
handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode='w')
handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
logger.addHandler(handler)


cogs = [maintenance, moderation, gambling, fun, music]

def get_prefix(client, message):
    with open('prefixes.json', 'r') as f:
        prefixes = json.load(f)
    return prefixes[str(message.guild.id)]

client = commands.Bot(command_prefix = get_prefix)
client.remove_command('help')

for i in range(len(cogs)):
    cogs[i].setup(client)

client.load_extension('dismusic')

client.lava_nodes = [
    {
        'host': 'lava.link',
        'port': 80,
        'rest_uri': f'http://lava.link:80',
        'identifier': 'MAIN',
        'password': 'anything',
        'region': 'rotterdam'
    }
]

client.run(TOKEN)

extensions/fun.py
import discord
from discord.ext import commands


class fun(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.command(aliases=['baum'])
    async def tree(self, ctx):
        with open('extensions/pictures/tree.png', 'rb') as f:
            picture = discord.File(f)
        await ctx.channel.send(file=picture)

    @commands.command(aliases=['fube'])
    async def gay(self, ctx):

        dev_id = 406540553262727180
        user_id = '610902170770079746'
        if ctx.message.author.id == dev_id:
            await ctx.channel.send(f'Gayest man alive <@{user_id}>\n https://scp-wiki.wikidot.com/scp-230')
            print(f'{ctx.message.author.id}')
        else:
            await ctx.message.channel.send('**Only the developer is able to use this command!**')

    @commands.command(aliases=['YEET'])
    async def yeet(self, ctx):
        with open('extensions/pictures/yeet.png', 'rb') as f:
            picture = discord.File(f)
        await ctx.channel.send(file=picture)
        await ctx.channel.send('YEET!')

def setup(client):
    client.add_cog(fun(client))


extensions/music.py
import discord
from discord.ext import commands

class music(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.command()
    async def music(self, ctx):
        mEmbed = discord.Embed(title="Music help", description="Music commands", color=0xfc1703)
        mEmbed.set_author(name=self.client.user.name, icon_url=str(self.client.user.avatar_url))
        mEmbed.add_field(name="Connect", value="s!connect | s!con [Connect the bot]", inline=False)
        mEmbed.add_field(name="Play", value="s!play (Name or link) [Play a song]", inline=False)
        mEmbed.add_field(name="Pause", value="s!pause [Pause the song]", inline=False)
        mEmbed.add_field(name="Resume", value="s!resume [Resume the song]", inline=False)
        mEmbed.add_field(name="Queue", value="s!queue [Show current queue]", inline=False)
        mEmbed.add_field(name="Now playing", value="s!nowplaying [Show current songs]", inline=False)
        mEmbed.add_field(name="Loop", value="s!loop [Loop current song or playlist]", inline=False)  
        mEmbed.add_field(name="Equalizer", value="s!equalizer [Equalizer]", inline=False)
        mEmbed.add_field(name="Seek", value="s!seek (seconds) [Seek player]", inline=False)              
        mEmbed.add_field(name="Disconnect", value="s!disconnext | s!dc [Disconnect bot]", inline=False)
        mEmbed.set_footer(text="Music extension:\n@Dismusic(pypi.org/project/dismusic/)")
        await ctx.message.channel.send(embed=mEmbed)

def setup(client):
    client.add_cog(music(client))


extensions/gambling.py
import discord
from discord.ext import commands
import random

class gambling(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.command()
    async def gambling(self, ctx):
        gEmbed = discord.Embed(title="Gambling help", description="Commands for gambling", color=0xfc1703)
        gEmbed.set_author(name=self.client.user.name, icon_url=str(self.client.user.avatar_url))
        gEmbed.add_field(name="8ball", value="s!8ball (question) [Watch into the Future]", inline=False)
        gEmbed.add_field(name="Coinflip", value="s!coinflip [Flip a coin]", inline=False)
        gEmbed.set_footer(text="Developed by Senpаi#0001")
        await ctx.message.channel.send(embed=gEmbed)

    @commands.command(aliases=['8ball'])
    async def _8ball(self, ctx, *, question):
        responses = ["It is certain.",
                "It is decidedly so.",
                "Without a doubt.",
                "Yes - definitely.",
                "You may rely on it.",
                "As I see it, yes.",
                "Most likely.",
                "Outlook good.",
                "Yes.",
                "Signs point to yes.",
                "Reply hazy, try again.",
                "Ask again later.",
                "Better not tell you now.",
                "Cannot predict now.",
                "Concentrate and ask again.",
                "Don't count on it.",
                "My reply is no.",
                "My sources say no.",
                "Outlook not so good.",
                "Very doubtful."]
        await ctx.send(f'Question: {question}\nAnswer: {random.choice(responses)}')

    @commands.command()
    async def coinflip(self, ctx):
        responses = ['heads','tails']
        await ctx.send(f'{random.choice(responses)}')

def setup(client):
    client.add_cog(gambling(client))

extensions/moderation.py
import discord
from discord.ext import commands

class moderation(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.command()
    async def moderation(self, ctx):
        mEmbed = discord.Embed(title="Moderation help", description="Commands for moderation", color=0xfc1703)
        mEmbed.set_author(name=self.client.user.name, icon_url=str(self.client.user.avatar_url))
        mEmbed.add_field(name="Clear", value="s!clear (value) [Clear the chat]", inline=False)
        mEmbed.add_field(name="Change prefix", value="s!changeprefix [Change the command prefix]", inline=False)
        mEmbed.add_field(name="Kick", value="s!kick (Name) [Kick a member]", inline=False)
        mEmbed.add_field(name="Ban", value="s!ban (Name) [Ban a member]", inline=False)
        mEmbed.add_field(name="Unban", value="s!unban (Name#????) [Ban a member]", inline=False)
        mEmbed.set_footer(text="Developed by Senpаi#0001")
        await ctx.message.channel.send(embed=mEmbed)

    @commands.command()
    @commands.has_permissions(manage_messages=True)
    async def clear(self, ctx, amount=5):
	    await ctx.channel.purge(limit=amount + 1)
            
    @commands.command()
    @commands.has_permissions(kick_members=True)
    async def kick(self, ctx, member : discord.Member, *, reason=None):
        await member.kick(reason=reason)

    @commands.command()
    @commands.has_permissions(ban_members=True)
    async def ban(self, ctx, member : discord.Member, *, reason=None):
        await member.ban(reason=reason)
        await ctx.send(f'Banned {member.mention}')

    @commands.command()
    @commands.has_permissions(ban_members=True)
    async def unban(self, ctx, *,member):
        banned_user = await ctx.guild.bans()

        member_name, member_discriminator = member.split('#')
        for ban_entry in banned_user:
            user = ban_entry.user

            if (user.name, user.discriminator) == (member_name, member_discriminator):
                await ctx.guild.unban(user)
                await ctx.send(f'Unbanned {user.mention}')
                return

def setup(client):
    client.add_cog(moderation(client))

extensions/maintenance.py
import os
import discord
from discord.ext import commands
import json


class maintenance(commands.Cog):
    def __init__(self, client):
        self.client = client

    #on ready
    @commands.Cog.listener()
    async def on_ready(self):
        await self.client.change_presence(activity=discord.Game('s!help'))
        print('Logged in as', self.client.user.name,'!')

    #on member join
    @commands.Cog.listener()
    async def on_member_join(member):
        print(f'{member} has joined a server')

    #on member remove
    @commands.Cog.listener()
    async def on_member_remove(member):
        print(f'{member} has left a server')

    #error handler
    @commands.Cog.listener()
    async def on_command_error(self, ctx: commands.Context, error: commands.CommandError):
        if isinstance(error, commands.CommandOnCooldown):
            message = f"This command is on cooldown. Please try again after {round(error.retry_after, 1)} seconds."
        elif isinstance(error, commands.MissingPermissions):
            message = "You are missing the required permissions to run this command!"
        elif isinstance(error, commands.MissingRequiredArgument):
            message = f"Missing a required argument: {error.param}"
        else:
            message = "Oh no! Something went wrong while running the command!"

        await ctx.send(message, delete_after=5)
        await ctx.message.delete(delay=5)

    #prefix change
    @commands.Cog.listener()
    async def on_guild_join(guild):
        with open('prefixes.json', 'r') as f:
            prefixes = json.load(f)

        prefixes[str(guild.id)] = 's!'

        with open('prefixes.json', 'w') as f:
            json.dump(prefixes, f, indent=4)

    @commands.Cog.listener()
    async def on_guild_remove(guild):
        with open('prefixes.json', 'r') as f:
            prefixes = json.load(f)

        prefixes.pop(str(guild.id))

        with open('prefixes.json', 'w') as f:
            json.dump(prefixes, f, indent=4)

    @commands.command()
    @commands.has_permissions(administrator=True)
    async def changeprefix(self, ctx, prefix):
        with open('prefixes.json', 'r') as f:
            prefixes = json.load(f)

        prefixes[str(ctx.guild.id)] = prefix

        with open('prefixes.json', 'w') as f:
            json.dump(prefixes, f, indent=4)

        await ctx.send(f'Prefix changed to: {prefix}')
        name=f'{prefix}BotBot'

    #ping
    @commands.command()
    async def ping(self, ctx):
        await ctx.send(f'Ping = {round(self.client.latency * 1000)}ms')

    #standart help
    @commands.command()
    async def help(self, ctx):
        hEmbed = discord.Embed(title="Help", description="Bot commands", color=0xfc1703)
        hEmbed.set_author(name=self.client.user.name, icon_url=str(self.client.user.avatar_url))
        hEmbed.add_field(name="Ping", value="s!ping [Shows the Bot Latency]", inline=False)
        hEmbed.add_field(name="Version", value="s!version [Shows the Bot version]", inline=False)
        hEmbed.add_field(name="Music", value="s!music [Music commands]", inline=False)
        hEmbed.add_field(name="Moderation", value="s!moderation [Moderation commads]", inline=False)
        hEmbed.add_field(name="Gambling", value="s!gambling [Mambling commands]", inline=False)
        hEmbed.add_field(name="Tree", value="s!tree / s!baum [Posts a picture of a Tree]", inline=False)
        hEmbed.set_footer(text="Developed by Senpаi#0001")
        await ctx.message.channel.send(embed=hEmbed)

    #version
    @commands.command()
    async def version(self,ctx):
        vEmbed = discord.Embed(title="Current Version", color=0xfc1703)
        vEmbed.set_author(name=self.client.user.name, icon_url=str(self.client.user.avatar_url))
        vEmbed.add_field(name="Bot version:", value="v.1.1", inline=False)
        vEmbed.add_field(name="Bot status:", value="work in progress", inline=False)
        vEmbed.add_field(name="Relesed", value="November 12th, 2021")
        vEmbed.set_footer(text="Developed by Senpаi#0001")
        await ctx.message.channel.send(embed=vEmbed)

def setup(client):
    client.add_cog(maintenance(client))
